### DISCUSSION TITLE
How to trigger an event after a download?

### DISCUSSION THREAD

**Discussion by Ryogin** [AutomaApp/automa#173] [Q&A]
**Short Question:** is there any way to trigger an event after a "download"?

**Long Question:** 
I work with some supplier websites in my country that allow me to download the updated product stock in CSV. AUTOMA allows me to enter each category and download the corresponding file, but since the servers are slow it takes several seconds for the file to be generated through the site's API. Then when is loading finished, AUTOMA does not continue with the other tasks, it just stays stuck and stops after minutes for "Timeout".

A logical solution would be that the following event will be trigged after the download, but I cannot find this option in AUTOMA.

Another would be to make an auto "hold" for the page to finish loading the API, and then trigger the next step. I can't find this option either.

Also, I don't want to put a delay between the triggers, because delay time varies a lot depending on the server load. I would have to put something like 1 minute, and for most of the time it would be much more than necessary.

Any other idea?

![image](https://user-images.githubusercontent.com/95941862/145649679-287e28a0-67a1-44e9-853a-54894213fa45.png)

**Comment by Kholid060** [comment]
If there's an element that indicates the file is to be generating, you can use the Element exists block to do something like the image below.

![image](https://user-images.githubusercontent.com/22908993/145675518-79931bb6-cda6-44d6-bbed-3a504c72180e.png)

When the element that indicates the file is generating still exists, it will continue to the delay block and then loop itself. And when it doesn't exist it will continue to the new tab block.
